src/string_bytes.h

```diff
@@ -12,6 +14,51 @@ extern int WRITE_UTF8_FLAGS;
 
 class StringBytes {
  public:
+  class InlineDecoder {
+   public:
+    explicit InlineDecoder(Environment* env) : env_(env), out_(nullptr) {
+    }
+
+    ~InlineDecoder() {
+      if (out_ != out_st_)
+        delete[] out_;
+      out_ = nullptr;
+    }
+
+    inline bool Decode(v8::Handle<v8::String> string,
+                       v8::Handle<v8::Value> encoding,
+                       enum encoding _default) {
+      enum encoding enc = ParseEncoding(env_->isolate(), encoding, _default);
+      if (!StringBytes::IsValidString(env_->isolate(), string, enc)) {
+        env_->ThrowTypeError("Bad input string");
+        return false;
+      }
+
+      size_t buflen = StringBytes::StorageSize(env_->isolate(), string, enc);
+      if (buflen > sizeof(out_st_))
+        out_ = new char[buflen];
+      else
+        out_ = out_st_;
+      size_ = StringBytes::Write(env_->isolate(),
+                                 out_,
+                                 buflen,
+                                 string,
+                                 enc);
+      return true;
+    }
+
+    inline const char* out() const { return out_; }
+    inline size_t size() const { return size_; }
+
+   private:
+    static const int kStorageSize = 1024;
+
+    Environment* env_;
```

Comment:
Maybe pass the env as an argument to Decode()?  It's not a big win but it reduces the stack footprint by a word.